src/api/entities/Identity/AssetPermissions.ts

```diff
@@ -116,7 +119,15 @@ export class AssetPermissions extends Namespace<Identity> {
 
     return P.map(assetEntries, async ([key]) => {
       const assetId = assetIdToString(key.args[1]);
-      const asset = new FungibleAsset({ assetId }, context);
+      let asset: FungibleAsset | NftCollection;
+      const typedAsset = await asAsset(assetId, context);
+
+      if (isFungibleAsset(typedAsset)) {
+        asset = new FungibleAsset({ assetId }, context);
+      } else {
+        asset = new NftCollection({ assetId }, context);
+      }
+
```

Comment:
The point of the helper function is to make this logic unneeded in other places. The runtime object is going to be one or the other without the if and extra construction

```ts
 const asset = await asAsset(assetId, context);
```